src/renderer/components/SettingsView/SettingsMenu.js

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react'
+import Tabs from '@material-ui/core/Tabs'
+import Tab from '@material-ui/core/Tab'
+import { useIntl } from 'react-intl'
+import ChevronRightIcon from '@material-ui/icons/ChevronRight'
+import styled from 'styled-components'
+import { Paper, Typography, useTheme } from '@material-ui/core'
+
+/**
+ * @typedef {object} tab
+ * @property {string} tab.tabId
+ * @property {string | import('react-intl').MessageDescriptor} tab.label
+ * @property {string | import('react-intl').MessageDescriptor} tab.subtitle
+ * @property {import('@material-ui/core/OverridableComponent').OverridableComponent<import('@material-ui/core').SvgIconTypeMap<{}, "svg">>} tab.icon
+ *
+ * @typedef tabs
+ * @property {tab[]} tabs
+ *
+ * @typedef currentTab
+ * @property {number} currentTab
+ *
+ * @typedef onTabChange
+ * @property {(e: React.Dispatch<number>) => number} onTabChange
+ *
+ * @typedef {tabs & currentTab & onTabChange} Props
+ */
+/** @param {Props} props */
+export const SettingsMenu = ({ tabs, currentTab, onTabChange }) => {
```

Comment:
just a slight refactor that will make the setting menu options typesafe:

```ts
/** @typedef {'AboutMapeo'} tabId */

/**
 * @typedef {object} tab
 * @property {tabId} tabId //We are adding all the menu options above, and setting this to only be the menu options instead of a generic string (which is just the one for now).
 * @property {string | import('react-intl').MessageDescriptor} tab.label
 * @property {string | import('react-intl').MessageDescriptor} tab.subtitle
 * @property {import('@material-ui/core/OverridableComponent').OverridableComponent<import('@material-ui/core').SvgIconTypeMap<{}, "svg">>} tab.icon
 *
 * @typedef tabs
 * @property {tab[]} tabs
 *
 * @typedef currentTab
 * @property {number} currentTab
 *
 * @typedef onTabChange
 * @property {(e: React.Dispatch<tabId | null>) => void} onTabChange //same thing as above, onTabChange doesn't take a generic string, it takes on the tabIds
 *
 * @typedef {tabs & currentTab & onTabChange} Props
 */
/** @param {Props} props */

```